factory/processFactory.go

```diff
@@ -79,3 +81,33 @@ func getPubKeyConverter(cfg config.GeneralConfig) (core.PubkeyConverter, error)
 		return nil, common.ErrInvalidPubKeyConverterType
 	}
 }
+
+// CreatePayloadHandler will create a new instance of payload handler
+func CreatePayloadHandler(connType string, marshaller marshal.Marshalizer, wsMarshaller marshal.Marshalizer, facade process.EventsFacadeHandler) (websocket.PayloadHandler, error) {
+	switch connType {
+	case common.WSObsConnectorType:
+		return createPayloadHandler(wsMarshaller, facade)
+	case common.HTTPConnectorType:
+		return createPayloadHandler(marshaller, facade)
```

Comment:
Is it ok that you create the same instance with different marshaller for both connections?

Reply:
refactored this part